### PR TITLE
beehive/context: fix the unlocking of the ctx.chsLock in the delChannel

### PR DESCRIPTION
**Pull request by Iceber**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix the unlocking of the ctx.chsLock in the delChannel
```

### CODE REVIEW

**Comment by Iceber** [comment]
Is it significant to print a warning log when a module does not exist in the `ctx.channels`?

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_channel.go [hunk 1]
```diff
@@ -308,13 +308,12 @@ func (ctx *ChannelContext) addChannel(module string, moduleCh chan model.Message
 func (ctx *ChannelContext) delChannel(module string) {
 	// delete module channel from channels map
 	ctx.chsLock.Lock()
-	_, exist := ctx.channels[module]
-	if !exist {
+	if _, exist := ctx.channels[module]; !exist {
+		ctx.chsLock.Unlock()
 		klog.Warningf("Failed to get channel, module:%s", module)
```
Comment (Iceber) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_channel.go:313]: Is it significant to print a warning log when a module does not exist in the `ctx.channels`?

or change:
```go
ctx.chsLock.Lock()
delete(ctx.channels, module)
ctx.chsLock.Unlock()
```

**Review by fisherxu — COMMENTED** [review]
How about using defer to unlock?

**Comment by Iceber** [comment]
> How about using defer to unlock?

@fisherxu There will be no subsequent operation on the `ctx.channels`, so there is no need to keep the`ctx.chsLock` locked.
And a lock will be placed on `typeChsLock` later, which may cause deadlocks if defer is used.

**Comment by fisherxu** [comment]
> > How about using defer to unlock?
> 
> @fisherxu There will be no subsequent operation on the `ctx.channels`, so there is no need to keep the`ctx.chsLock` locked.
> And a lock will be placed on `typeChsLock` later, which may cause deadlocks if defer is used.

Oh, sounds good, had ignored the `typeChsLock` below..
/lgtm

**Pull request merged**